prog/hints.go

```diff
@@ -105,10 +105,12 @@ func generateHints(compMap CompMap, arg Arg, exec func()) {
 
 func checkConstArg(arg *ConstArg, compMap CompMap, exec func()) {
 	original := arg.Val
+	sizeLimit := uint64(1<<(8*arg.Size()) - 1)
 	// Note: because shrinkExpand returns a map, order of programs is non-deterministic.
 	// This can affect test coverage reports.
 	for _, replacer := range shrinkExpand(original, compMap) {
-		arg.Val = replacer
+		// Truncates the values that are too large.
+		arg.Val = replacer & sizeLimit
```

Comment:
Simply truncating here won't do.
We also want to exclude some pairs where (1) we can prove that the first value couldn't come from the argument due to size, (2) where the second value won't fit into the argument due to size.

Reply:
See the issue description https://github.com/google/syzkaller/issues/507#issue-288894818